backend/open_webui/models/chats.py

```
@@ -904,5 +907,148 @@ def delete_shared_chats_by_user_id(self, user_id: str) -> bool:
+
+                # Apply age filter if specified
+                if max_age_days is not None:
+                    import time
+
+                    cutoff_timestamp = int(time.time()) - (max_age_days * 24 * 60 * 60)
+                    query = query.filter(Chat.created_at < cutoff_timestamp)
+
+                # Apply preservation filters
+                if preserve_pinned:
+                    query = query.filter(
+                        or_(Chat.pinned == False, Chat.pinned.is_(None))
```

Ruff (0.14.0)

Avoid equality comparisons to `False`; use `not Chat.pinned:` for false checks

Replace with `not Chat.pinned`

(E712)